test/ModuleManagerTest.php

```diff
@@ -100,6 +100,34 @@ public function testModuleLoadingBehavior()
         self::assertSame(1, count($modules));
     }
 
+    public function testModuleLoadingBehaviorWithModuleClassStrings()
+    {
+        $moduleManager = new ModuleManager(['SomeModule'], $this->events);
+        $this->defaultListeners->attach($this->events);
+        $modules = $moduleManager->getLoadedModules();
+        self::assertSame(0, count($modules));
+        $modules = $moduleManager->getLoadedModules(true);
+        self::assertSame(1, count($modules));
+        $moduleManager->loadModules(); // should not cause any problems
+        $moduleManager->loadModule(Module::class); // should not cause any problems
+        $modules = $moduleManager->getLoadedModules(true); // BarModule already loaded so nothing happens
+        self::assertSame(1, count($modules));
+    }
+
+    public function testModuleLoadingBehaviorWithModuleClassStringsVersion2()
+    {
+        $moduleManager = new ModuleManager([Module::class], $this->events);
+        $this->defaultListeners->attach($this->events);
+        $modules = $moduleManager->getLoadedModules();
+        self::assertSame(0, count($modules));
+        $modules = $moduleManager->getLoadedModules(true);
+        self::assertSame(1, count($modules));
+        $moduleManager->loadModules(); // should not cause any problems
+        $moduleManager->loadModule('SomeModule'); // should not cause any problems
+        $modules = $moduleManager->getLoadedModules(true); // BarModule already loaded so nothing happens
+        self::assertSame(1, count($modules));
+    }
```

Comment:
Please use a data provider here and add some blank lines to break the wall of code / text.